Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -223,8 +235,17 @@ class TrackStateProxy {
       auto dest = getMask();
       auto src = other.getMask() &
                  mask;  // combine what we have with what we want to copy
-      if (static_cast<std::underlying_type_t<TrackStatePropMask>>((src ^ dest) &
-                                                                  src) != 0) {
+
+      if (ACTS_CHECK_BIT(src, PM::Calibrated)) {
+        // on-demand allocate calibrated
+        dest |= PM::Calibrated;
+      }
+
+      if ((static_cast<std::underlying_type_t<TrackStatePropMask>>(
+               (src ^ dest) & src) != 0 ||
+           dest == TrackStatePropMask::None ||
+           src == TrackStatePropMask::None) &&
+          mask != TrackStatePropMask::None) {
```

Comment:
Can any hint of what is wrong be part of exception message (the requirements are not quite nontrivial).

Reply:
Also, what should happen if destination TS has smaller size allocated for measurement?

Reply:
I'll add some comments to the if to explain what it tries to check.

If the destination TS has a calibrated measurement allocated, it will reallocated **unless** the dimensions match up. So even if we copy from smaller to larger we will reallocate. My thinking was that in almost all cases we will be allocating from nothing, so it doesn't make a difference.